File: example01/better-code-1.ts
function createValidator(
  number: number,
  mode: string = "Max"
): ((value: number) => boolean) | undefined {
  if (mode === "Max") {
    return (value: number) => value < number;
  } else if (mode === "Min") {
    return (value: number) => value > number;
  }
}

const isBelowMax = createValidator(31);

console.log(isBelowMax!(15));
console.log(isBelowMax!(32));

File: example01/better-code-1.py
def create_validator (mode,number):
    if mode == 'Max':
        return lambda value: value < number
    elif mode == 'Min': 
        return lambda value: value > number
    
is_below_max = create_validator("Max", 31)

print(is_below_max(15))
print(is_below_max(32))